tools/build_api.py

```diff
@@ -230,11 +231,15 @@ def find_valid_toolchain(target, toolchain):
     """
     end_warnings = []
     toolchain_names = get_valid_toolchain_names(target, toolchain)
+    if(target.default_toolchain == "uARM"):
```

Comment:
```suggestion
    if target.default_toolchain == "uARM":
```
Why was it necessary to add the check for `target.default_toolchain`? What happens if it is not there?

Reply:
for some targets device only have default_toolchain with uARM so added this check for the targets only build with uARM to append deprecate message